hack/generated/controllers/controller_resources.go

```diff
@@ -7,19 +7,20 @@ package controllers
 
 import (
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	resources "github.com/Azure/azure-service-operator/hack/generated/_apis/microsoft.resources/v1alpha1api20200601"
 )
 
-func GetKnownStorageTypes() []runtime.Object {
+func GetKnownStorageTypes() []client.Object {
 	knownTypes := getKnownStorageTypes()
 
 	knownTypes = append(knownTypes, new(resources.ResourceGroup))
 
 	return knownTypes
 }
 
-func GetKnownTypes() []runtime.Object {
+func GetKnownTypes() []client.Object {
```

Comment:
`client.Object` is actually a combo of `runtime.Object` and `metav1.Object`. You can see them explain it [here](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.9.2/pkg/client/object.go#L45).

This is actually a nice win because we actually want to make sure all of our CRD types are `client.Object` (have both `runtime,Object` and `metav1.Object`), so using this everywhere helps ensure that.